src/components/wallets/userWalletComponent.tsx

```diff
@@ -0,0 +1,97 @@
+import { useEffect, useState } from "react";
+import { Button } from "../ui/button";
+import { useSiwe } from "@/app/(dashboard)/utils/useSiwe";
+import { useEthersSigner } from "@/lib/ethersUtils";
+import { useAccount } from "wagmi";
+import { ConnectButton } from "@rainbow-me/rainbowkit";
+import { useSession } from "next-auth/react";
+import { UserWallet } from "@prisma/client";
+import axios from "axios";
+import { Icons } from "../icons";
+import { Input } from "../ui/input";
+import { Skeleton } from "../ui/skeleton";
+
+export function UserWalletComponent() {
+  const { data: session } = useSession();
+  const [isLoading, setIsLoading] = useState(true);
+  const [userWallet, setUserWallet] = useState<UserWallet | null>(null);
+  const signer = useEthersSigner();
+  const account = useAccount();
+
+  const handleAddWallet = async () => {
+    if (signer) {
+      await useSiwe(
+        signer,
+        "Verify you own this address to add to the platform",
+      );
+      handleFetchWallet();
+    }
+  };
+
+  const handleFetchWallet = async () => {
+    const { data } = await axios.get(`/api/wallet`);
+    if (data.success) {
+      setUserWallet(data.wallet);
+    }
+  };
+
+  useEffect(() => {
+    handleFetchWallet();
+    setIsLoading(false);
+  }, [session]);
+
+  return (
+    <>
+      <div>
+        <h3 className="text-lg font-medium">Manage Your Wallet</h3>
+        <p className="text-sm text-muted-foreground">
+          This is your wallet profits and controbutions will be payedout too.
```

Comment:
typos

Reply:
"Register the wallet that you wish to use to receive payments and rewards from your contributions"